### PR TITLE
Click next generation

### PR DESCRIPTION
**Pull request by doutriaux1**
Ready for review

examples of new functionalities can be seen at:
https://github.com/PCMDI/pcmdi_metrics/tree/portraits_jupyter/doc/jupyter

also in baselines at: https://github.com/UV-CDAT/uvcdat-testdata/pull/177

### CODE REVIEW

**Comment by durack1** [comment]
@doutriaux1 this is great, looking forward to pulling this into our analysis pipeline and getting these new features live on the webserver.

@danlipsa @aashish24 when can you folks take a look and review this branch? It's something that we're really keen to starting working on ASAP

@crean2 @gleckler1 ping

**Comment by doutriaux1** [comment]
@danlipsa ping

**Comment by doutriaux1** [comment]
@james_crean

**Comment by lee1043** [comment]
@doutriaux1 Thanks for adding this. Can't wait to see it merged!

**Comment by durack1** [comment]
@doutriaux1 @James-Crean I'm also looking forward to seeing this in action, thanks!

**Comment by gleckler1** [comment]
Yea!  We'll give it a go monday morning!

**Comment by durack1** [comment]
@James-Crean please ping us a link (via email) when you have a new prototype page live - this is gonna be great!

**Comment by gleckler1** [comment]
@James-Crean @doutriaux1 @durack1  JC thanks!! for the example https://oceanonly.llnl.gov/crean2/plots/new-modal/clickable_portrait_6.html

Although its not critical to get the idea, I can't see images I presume for permission reasons...

More importantly,  can you verify that with a modal we have webpage functionality, e.g., text and links in addition to image?  ... example:  https://oceanonly.llnl.gov/gleckler1/portraits/clickable_variability.html

Also, per PD's recommendation it does seem a good idea when hovering to be seeing lowres thumbnails.

File: vcs/clickMap.py [hunk 1]
```diff
@@ -5,22 +5,36 @@
...
+class Param(object):
+    def __init__(self, area, target):
+        self.area = area
+        self.target = target
+
+
+def createAreaTag(parameters):
+    """Create an area tag to go along with an html map tag
+    input: parameters object
+    At the minimum parameter object needs thew followin g two attributes:
```
Comment (durack1) [inline vcs/clickMap.py]: Typo `followin g`

**Comment by durack1** [comment]
@gleckler1 the low-res thumbnails will need to be generated by us at the same time that the maps are generated. @James-Crean is it easy to ascertain what thumbnail sizes would be useful to generate considering the standard modal template sizes? If we're going to optimize things, may as well start from that info first

File: vcs/Canvas.py [hunk 2]
```diff
@@ -2355,16 +2368,79 @@ def gettextextent(self, textobject):
                 [[...]]
 
         :param textobject: A VCS text object
+        :param angle: If not None overwrites the textobject's angle (in degree)
```
Comment (durack1) [inline vcs/Canvas.py]: `degree` - > `degrees`

File: vcs/clickMap.py [hunk 3]
```diff
@@ -52,10 +66,19 @@ def mapPng(image, areas, targets=[], tooltips=[], width=None, height=None, name=
     :param areas: list of each polygon coordinate on the image
     :type areas: `list`_
 
-    :param targets: list of target URL for each map area. List will be completed with '#'s to match length of areas
+    :param targets: list of target URL for each area tag.List will be completed with '#'s to match length of areas
```
Comment (durack1) [inline vcs/clickMap.py]: `tag.List` -> `tag. List`

**Comment by James-Crean** [comment]
@gleckler1, There are no images to display simply because it was quicker to not generate them and transfer them. I added a single image to the first column, second row so that you could see what a populated one would look like. 

As long as we can agree on a consistent naming scheme for the data-* attributes, we can add lots of different things to the modal including arbitrary html under the picture. 

@durack1: 200px x 154.53px is the size of the tooltip image. Anything higher than 200px wide and the browser will scale it down to fit, so I would say generate them at 200px wide while keeping the aspect ratio. They shouldn't lose any quality, but they will be smaller.

**Comment by durack1** [comment]
@James-Crean ok great, 200px width is a great target. I'll start working toward updating my map plots so that thumbnails (@200px) are also produced

**Comment by gleckler1** [comment]
@James-Crean @doutriaux1 @durack1  HNY guys!   Where should I begin to try out the modal portrait code?

**Comment by doutriaux1** [comment]
@danlipsa please review this one as well, I really need this in soon.

**Comment by doutriaux1** [comment]
@aashish24 you can review too, it's a bigger PR so more reviewer is better.

**Comment by doutriaux1** [comment]
@aashish24 @danlipsa ping!

**Pull request merged**